src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/appsettings.json

```diff
@@ -0,0 +1,9 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+    }
+  },
+  "AllowedHosts": "*"
```

Comment:
This can be removed from this template as it's for the HTTPS redirection middleware.

Reply:
Doing a quick check of the code, it looks to me that this is for "host filtering" middleware:

https://github.com/dotnet/aspnetcore/blob/d07f377b6c4c72f44a0eb2810c8abf1ae3fabd40/src/DefaultBuilder/src/WebHost.cs#L253-L268

https://github.com/dotnet/aspnetcore/blob/d07f377b6c4c72f44a0eb2810c8abf1ae3fabd40/src/Middleware/HostFiltering/src/HostFilteringMiddleware.cs#L63-L70

Although, that code says if there is no `AllowedHosts` configured, it uses `*` anyway. So this configuration is redundant. I'm just wondering why we have it in our web api template:

https://github.com/dotnet/aspnetcore/blob/d07f377b6c4c72f44a0eb2810c8abf1ae3fabd40/src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/appsettings.json#L64

Reply:
If we remove it from one template for being redundant, we should remove it from all. I'd prefer to leave it, and creating an issue to decide whether it should be removed everywhere.

Reply:
Hmmm, we might both be wrong, and this could have been intended for forwarded haders. I seem to recall we put it in the template to aid discovery as part of some security-minded effort, i.e. our recommendation would be for folks to actually configure this when running behind a reverse proxy/ingress that does header forwarding to ensure only headers from the expected hosts are forwarded @blowdart

Reply:
OK I guess we just leave it there then and we can create an issue to revisit it in all templates if desired.

Reply:
> Hmmm, we might both be wrong, and this could have been intended for forwarded haders. 

I'm not seeing any code that reads this from the configuration and sets it for forwarded headers:

https://source.dot.net/#Microsoft.AspNetCore.HttpOverrides/ForwardedHeadersOptions.cs,6f32cde727e8aa7d,references

I only see it being read from the config for host filtering.

Reply:
OK honestly I have no idea what it's there then. My recollection was that this was related to HTTPS redirection but if it's for host filtering then I'd question why it's there unless there's a very clear link from that config property to a reason for host filtering to be configured by our customers. The [doc on this is relatively small](https://learn.microsoft.com/aspnet/core/fundamentals/servers/kestrel/host-filtering) but it says that the config value has to exist for the host filtering middleware to be enabled by the default host, but the code @eerhardt posted above seems to indicate it just falls back to configuring this anyway.

Reply:
I've opened https://github.com/dotnet/aspnetcore/issues/47862 to consider changing all the templates.